pkg/controller/internal/harbor_commons.go

```diff
@@ -91,3 +97,24 @@ func CheckFilterType(filterType h.FilterType) error {
 	}
 	return errors.New("the provided filter type could not be checked")
 }
+
+func FetchReadyHarborInstance(ctx context.Context, namespace, parentInstance string, r client.Client) (*registriesv1alpha1.Instance, error) {
+	harbor := &registriesv1alpha1.Instance{}
+	ns := types.NamespacedName{
+		Namespace: namespace,
+		Name:      parentInstance,
+	}
+
+	err := r.Get(ctx, ns, harbor)
+	if apierrors.IsNotFound(err) {
+		return nil, ErrInstanceNotFound
+	} else if err != nil {
+		return nil, err
+	}
+
+	// Reconcile only if the corresponding harbor instance is in 'Ready' state
+	if harbor.Status.Phase.Name != registriesv1alpha1.InstanceStatusPhaseReady {
```

Comment:
Just to be sure: Are there any pointer properties in this chain (`harbor.Status` or `harbor.Status.Phase`) that might be `nil`?

Reply:
There are none :+1: 
`harbor.Status.Phase.LastTransition` is the only pointer type _in relation_ to this, though it is not called here.